pds_doi_core/actions/draft.py

```diff
@@ -97,11 +114,19 @@ def _transform_pds4_label_into_osti_record(self, input_file, contributor_value):
         if not input_file.startswith('http'):
             # Only process .xml files and print WARNING for any other files, then continue.
             if input_file.endswith('.xml'):
-                xml_tree = etree.parse(input_file)
+                try:
+                   xml_tree = etree.parse(input_file)
+                except OSError as e:
+                    msg = f'Error reading file {input_file}'
+                    logger.error(msg)
+                    raise InputFormatException(msg)
+
             else:
                 msg = f"Expecting .xml files only, encountering {input_file}"
-                logger.error(msg)
-                raise InputFormatException(msg)
+                logger.warning(msg)
```

Comment:
We can't raise an exception with behave testing since the input directory can containing many different kinds of files.  Raising an exception will cause the test to exit prematurely.

I don't know a happy solution.

Reply:
Yes that makes sense, warning is a perfect solution. Thanks